src/rust/src/wallet.rs

```diff
@@ -170,27 +322,53 @@ impl Wallet {
             .collect::<Vec<_>>();
         let mut result = vec![];
         for (action_idx, ivk, note, recipient, memo) in bundle.decrypt_outputs_for_keys(&keys) {
-            // Mark the witness tree with the fact that we want to be able to compute
-            // a witness for this note.
-            let note_data = DecryptedNote { note, memo };
+            let outpoint = OutPoint {
+                txid: *txid,
+                action_idx,
+            };
+
+            // Generate the nullifier for the received note and add it to the nullifiers map so
+            // that we can detect when the note is later spent.
+            if let Some(fvk) = self.key_store.viewing_keys.get(&ivk) {
+                let nf = note.nullifier(fvk);
+                // If we already have an entry for this nullifier that does not correspond
+                // to the current outpoint, then add the existing outpoint and the current
+                // outpoint to the conflicts map. (The existing output will already be
+                // present if this is not the first detected conflict for this nullifier.)
+                if let Some(op) = self.nullifiers.get(&nf) {
+                    if op != &outpoint {
```

Comment:
This comment and code is getting the notion of `conflicts` backwards.

A nullifier can only correspond to a single `OutPoint`, precisely because nullifiers are derived from output notes. That is, the `self.nullifiers` lookup will (computationally) never return an `OutPoint` that is not precisely equal to the current one.

The conflicts arise when a nullifier is _spent_ in multiple transactions, i.e. `conflicts` should be a map from `Nullifier` to in-points (the txid and index pointing to the action in which the nullifier appears, and its corresponding note is being spent). Use the `InPoint` type and move this logic to the nullifier-detecting part of bundle scanning.